src/fpylll/tools/probabilistic_bkz_simulator.py

```diff
@@ -0,0 +1,215 @@
+# -*- coding: utf-8 -*-
+"""
+BKZ simulation algorithm as proposed in
+
+- Bai, S., & Stehlé, D. & Wen, W. (2018). Measuring, simulating and exploiting
+  the head concavity phenomenon in BKZ. In T. Peyrin, & S. Galbraith,
+  ASIACRYPT~2018 : Springer, Heidelberg.
+
+The code is based on the implementation of the simulation algorithm proposed by
```

Comment:
Should this be here?

Reply:
What does "this" refer to?

Reply:
The "The code is based on" comment, it's based o a different simulator, no?

Reply:
By that comment I mean that I opened the CN11 simulator source in fpylll, and made the changes following the paper. Then I compared to the original implementation that Shi pointed me to after I had written their simulator.